### ISSUE TITLE
 ipv6  doesn't work anymore  after updating from 1.11.5 to 1.12.2

### ISSUE THREAD

**Issue by arianvp** [istio/istio#37410]
### Bug Description

When the pod only has an ipv6 address, the sidecar fails to start. It seems to try to create iptables rules in the ipv4 table which doesn't exist in this case.   This is a regression that was introduced in 1.12 as 1.11 worked fine so far

**Before updating  on version **1.11.5** the sidecards got created just fine**
```
% kubectl get pod -owide
NAME                     READY   STATUS    RESTARTS   AGE     IP                         NODE       NOMINATED NODE   READINESS GATES
nginx-85b98978db-hprkr   2/2     Running   0          2m10s   2001:19f0:6c01:1f10:2::7   worker-1   <none>           <none>
```
```
 % kubectl logs -l app=nginx --all-containers                
2022-02-17T14:38:21.334472Z     info    ads     XDS: Incremental Pushing:0 ConnectedEndpoints:2 Version:
2022-02-17T14:38:21.334569Z     info    cache   returned workload trust anchor from cache       ttl=23h59m59.665433537s
2022-02-17T14:38:21.334598Z     info    cache   returned workload trust anchor from cache       ttl=23h59m59.665404565s
2022-02-17T14:38:21.334851Z     info    cache   returned workload certificate from cache        ttl=23h59m59.665157079s
2022-02-17T14:38:21.335061Z     info    ads     SDS: PUSH request for node:nginx-85b98978db-hprkr.default resources:1 size:1.1kB resource:ROOTCA
2022-02-17T14:38:21.335124Z     info    cache   returned workload trust anchor from cache       ttl=23h59m59.664879103s
2022-02-17T14:38:21.335153Z     info    ads     SDS: PUSH for node:nginx-85b98978db-hprkr.default resources:1 size:1.1kB resource:ROOTCA
2022-02-17T14:38:21.335062Z     info    ads     SDS: PUSH request for node:nginx-85b98978db-hprkr.default resources:1 size:4.0kB resource:default
2022-02-17T14:38:22.594234Z     info    Initialization took 1.542331611s
2022-02-17T14:38:22.594259Z     info    Envoy proxy is ready
-A ISTIO_OUTPUT ! -d 127.0.0.1/32 -o lo -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
COMMIT
# Completed on Thu Feb 17 14:38:15 2022

2022-02-17T14:38:15.082858Z     info    Running command: ip6tables-save 
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/02/17 14:38:20 [notice] 1#1: using the "epoll" event method
2022/02/17 14:38:20 [notice] 1#1: nginx/1.21.6
2022/02/17 14:38:20 [notice] 1#1: built by gcc 10.2.1 20210110 (Debian 10.2.1-6) 
2022/02/17 14:38:20 [notice] 1#1: OS: Linux 5.10.84-flatcar
2022/02/17 14:38:20 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2022/02/17 14:38:20 [notice] 1#1: start worker processes
2022/02/17 14:38:20 [notice] 1#1: start worker process 31
2022/02/17 14:38:20 [notice] 1#1: start worker process 32
arian@Arians-MBP vultr-kubernetes % kubectl logs -l app=nginx --all-containers
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/02/17 14:38:20 [notice] 1#1: using the "epoll" event method
2022/02/17 14:38:20 [notice] 1#1: nginx/1.21.6
2022/02/17 14:38:20 [notice] 1#1: built by gcc 10.2.1 20210110 (Debian 10.2.1-6) 
2022/02/17 14:38:20 [notice] 1#1: OS: Linux 5.10.84-flatcar
2022/02/17 14:38:20 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2022/02/17 14:38:20 [notice] 1#1: start worker processes
2022/02/17 14:38:20 [notice] 1#1: start worker process 31
2022/02/17 14:38:20 [notice] 1#1: start worker process 32
2022-02-17T14:38:21.334472Z     info    ads     XDS: Incremental Pushing:0 ConnectedEndpoints:2 Version:
2022-02-17T14:38:21.334569Z     info    cache   returned workload trust anchor from cache       ttl=23h59m59.665433537s
2022-02-17T14:38:21.334598Z     info    cache   returned workload trust anchor from cache       ttl=23h59m59.665404565s
2022-02-17T14:38:21.334851Z     info    cache   returned workload certificate from cache        ttl=23h59m59.665157079s
2022-02-17T14:38:21.335061Z     info    ads     SDS: PUSH request for node:nginx-85b98978db-hprkr.default resources:1 size:1.1kB resource:ROOTCA
2022-02-17T14:38:21.335124Z     info    cache   returned workload trust anchor from cache       ttl=23h59m59.664879103s
2022-02-17T14:38:21.335153Z     info    ads     SDS: PUSH for node:nginx-85b98978db-hprkr.default resources:1 size:1.1kB resource:ROOTCA
2022-02-17T14:38:21.335062Z     info    ads     SDS: PUSH request for node:nginx-85b98978db-hprkr.default resources:1 size:4.0kB resource:default
2022-02-17T14:38:22.594234Z     info    Initialization took 1.542331611s
2022-02-17T14:38:22.594259Z     info    Envoy proxy is ready
-A ISTIO_OUTPUT ! -d 127.0.0.1/32 -o lo -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
COMMIT
# Completed on Thu Feb 17 14:38:15 2022
```

**After upgrading to 1.12.2 the sidecar container doesn't start anymore and crashloopbackoffs**

```
% istioctl upgrade
% kubectl rollout restart deployment nginx
% kubectl get pod -owide
NAME                     READY   STATUS                  RESTARTS     AGE    IP                         NODE       NOMINATED NODE   READINESS GATES
nginx-556bc7cbd8-gq2pg   2/2     Running                 0            114s   2001:19f0:6c01:1f10:2::c   worker-1   <none>           <none>
nginx-6ddd56bf6-w68zx    0/2     Init:CrashLoopBackOff   1 (4s ago)   7s     2001:19f0:6c01:1f10:1::c   worker-0   <none>           <none>
```

```
% kubectl logs -l app=nginx --all-containers
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
COMMIT
2022-02-17T14:27:32.342611Z     info    Running command: iptables-restore --noflush /tmp/iptables-rules-1645108052342433684.txt1047550074
2022-02-17T14:27:32.343854Z     error   Command error output: xtables other problem: line 2 failed
2022-02-17T14:27:32.343869Z     error   Failed to execute: iptables-restore --noflush /tmp/iptables-rules-1645108052342433684.txt1047550074, exit status 1
```

### Version

```prose
% kubectl version --short
Client Version: v1.22.3
Server Version: v1.23.3
```

Before:

```
% istioctl version             
client version: 1.11.5
control plane version: 1.11.5
data plane version: 1.11.5 (2 proxies)
```

After:

```
% istioctl version
client version: 1.12.2
control plane version: 1.12.2
data plane version: 1.12.2 (1 proxies)
```

### Additional Information

**Before update:**

[bug-report-1.11.5.tar.gz](https://github.com/istio/istio/files/8089546/bug-report-1.11.5.tar.gz)
```
 % istioctl bug-report

Target cluster context: kubernetes-admin@kubernetes

Running with the following config: 

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-system, kube-public, kube-node-lease, local-path-storage } AND { Namespaces: kube-system, kube-public, kube-node-lease, local-path-storage }
end-time: 2022-02-17 15:46:34.09607 +0100 CET


The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.11.5", GitRevision:"4959f6f447280f5fd702f1af92ad26630fb00bd6", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.11.5"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.11.5}


Fetching proxy logs for the following containers:

default/nginx/nginx-85b98978db-ns278/istio-proxy
default/nginx/nginx-85b98978db-ns278/nginx
istio-system/istio-ingressgateway/istio-ingressgateway-86b75f74c9-j8kvq/istio-proxy
istio-system/istiod/istiod-7c8c747bd-sg8sq/discovery

Fetching Istio control plane information from cluster.

Running istio analyze on all namespaces and report as below:
Analysis Report:
Info [IST0102] (Namespace istio-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace istio-system istio-injection=enabled' to enable it, or 'kubectl label namespace istio-system istio-injection=disabled' to explicitly mark it as not needing injection.
Creating an archive at /Users/arian/Projects/vultr-kubernetes/bug-report.tar.gz.
Cleaning up temporary files in /var/folders/_t/r_qtl_3d1g7dm2vz75h1vx880000gn/T/bug-report.
Done.
```

**After update to 1.12.2**

[bug-report-1.12.2.tar.gz](https://github.com/istio/istio/files/8089616/bug-report-1.12.2.tar.gz)

```
% istioctl bug-report

Target cluster context: kubernetes-admin@kubernetes

Running with the following config: 

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-system,kube-public,kube-node-lease,local-path-storage }
end-time: 2022-02-17 15:54:39.382633 +0100 CET



Cluster endpoint: https://[2a05:f480:1800:263:5400:3ff:fedc:d9e9]:6443
CLI version:
version.BuildInfo{Version:"1.12.2", GitRevision:"unknown", GolangVersion:"go1.17.7", BuildStatus:"Nix", GitTag:"1.12.2"}

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.12.2", GitRevision:"af0d66fd0aa363e9a7b0164f3a94ba36252fe60f", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.12.2"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.12.2, 1.11.5}


Fetching proxy logs for the following containers:

default/nginx/nginx-556bc7cbd8-gq2pg/istio-proxy
default/nginx/nginx-556bc7cbd8-gq2pg/nginx
default/nginx/nginx-6ddd56bf6-w68zx/istio-proxy
default/nginx/nginx-6ddd56bf6-w68zx/nginx
istio-system/istio-ingressgateway/istio-ingressgateway-c6d9f449-n8kzk/istio-proxy
istio-system/istiod/istiod-5ffcccb477-v5pvd/discovery

Fetching Istio control plane information from cluster.

Running istio analyze on all namespaces and report as below:
Analysis Report:
Warning [IST0105] (Pod default/nginx-556bc7cbd8-gq2pg) The image of the Istio proxy running on the pod does not match the image defined in the injection configuration (pod image: docker.io/istio/proxyv2:1.11.5; injection configuration image: docker.io/istio/proxyv2:1.12.2). This often happens after upgrading the Istio control-plane and can be fixed by redeploying the pod.
Creating an archive at /Users/arian/Projects/vultr-kubernetes/bug-report.tar.gz.
Cleaning up temporary files in /var/folders/_t/r_qtl_3d1g7dm2vz75h1vx880000gn/T/bug-report.
```

**Comment by howardjohn** [comment]
There were a LOT of ipv6 PRs in this time frame. Mostly by me..

Do you have the full logs of the init contianer? seems the bug report does not/did not capture it

**Comment by arianvp** [comment]
That's comforting news! Happy to try it out on this cluster even if they didnt end up in a release yet. Just need to point me to a `istioctl` binary I can use.

Here is the entire log for the init container:

```
% kubectl create deployment nginx --image nginx
% kubectl logs -lapp=nginx -c istio-init
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
COMMIT
2022-02-17T18:56:56.188299Z     info    Running command: iptables-restore --noflush /tmp/iptables-rules-1645124216187955375.txt1906670297
2022-02-17T18:56:56.190020Z     error   Command error output: xtables other problem: line 2 failed
2022-02-17T18:56:56.190131Z     error   Failed to execute: iptables-restore --noflush /tmp/iptables-rules-1645124216187955375.txt1906670297, exit status 1
```

**Comment by howardjohn** [comment]
I think when you use `-l` for logs it only gets 10 logs. Either hit the pod directly or add `--tail=100000` or something will get more info. And I meant there were a lot of PRs in 1.11...1.12 for iptables+ipv6. So likely one of them broke something

**Comment by arianvp** [comment]
```
 % kubectl logs nginx-85b98978db-zdzx2 -c istio-init --tail 1000000
2022-02-17T19:32:51.208285Z     info    Istio iptables environment:
ENVOY_PORT=
INBOUND_CAPTURE_PORT=
ISTIO_INBOUND_INTERCEPTION_MODE=
ISTIO_INBOUND_TPROXY_ROUTE_TABLE=
ISTIO_INBOUND_PORTS=
ISTIO_OUTBOUND_PORTS=
ISTIO_LOCAL_EXCLUDE_PORTS=
ISTIO_EXCLUDE_INTERFACES=
ISTIO_SERVICE_CIDR=
ISTIO_SERVICE_EXCLUDE_CIDR=
ISTIO_META_DNS_CAPTURE=
INVALID_DROP=

2022-02-17T19:32:51.208348Z     info    Istio iptables variables:
PROXY_PORT=15001
PROXY_INBOUND_CAPTURE_PORT=15006
PROXY_TUNNEL_PORT=15008
PROXY_UID=1337
PROXY_GID=1337
INBOUND_INTERCEPTION_MODE=REDIRECT
INBOUND_TPROXY_MARK=1337
INBOUND_TPROXY_ROUTE_TABLE=133
INBOUND_PORTS_INCLUDE=*
INBOUND_PORTS_EXCLUDE=15090,15021,15020
OUTBOUND_IP_RANGES_INCLUDE=*
OUTBOUND_IP_RANGES_EXCLUDE=
OUTBOUND_PORTS_INCLUDE=
OUTBOUND_PORTS_EXCLUDE=
KUBE_VIRT_INTERFACES=
ENABLE_INBOUND_IPV6=true
DNS_CAPTURE=false
DROP_INVALID=false
CAPTURE_ALL_DNS=false
DNS_SERVERS=[],[]
OUTPUT_PATH=
NETWORK_NAMESPACE=
CNI_MODE=false
EXCLUDE_INTERFACES=

2022-02-17T19:32:51.208788Z     info    Writing following contents to rules file: /tmp/iptables-rules-1645126371208642827.txt944563614
* nat
-N ISTIO_INBOUND
-N ISTIO_REDIRECT
-N ISTIO_IN_REDIRECT
-N ISTIO_OUTPUT
-A ISTIO_INBOUND -p tcp --dport 15008 -j RETURN
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
-A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-ports 15006
-A PREROUTING -p tcp -j ISTIO_INBOUND
-A ISTIO_INBOUND -p tcp --dport 22 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15090 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15021 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15020 -j RETURN
-A ISTIO_INBOUND -p tcp -j ISTIO_IN_REDIRECT
-A OUTPUT -p tcp -j ISTIO_OUTPUT
-A ISTIO_OUTPUT -o lo -s 127.0.0.6/32 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --uid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
COMMIT
2022-02-17T19:32:51.208823Z     info    Running command: iptables-restore --noflush /tmp/iptables-rules-1645126371208642827.txt944563614
2022-02-17T19:32:51.210740Z     error   Command error output: xtables other problem: line 2 failed
2022-02-17T19:32:51.210754Z     error   Failed to execute: iptables-restore --noflush /tmp/iptables-rules-1645126371208642827.txt944563614, exit status 1
```

**Comment by howardjohn** [comment]
Hmm. I tried locally running the command on recent and 1.11 version and the output is exactly the same... not clear what has changed

**Comment by arianvp** [comment]
Do you mean you can't reproduce the non-zero exit code? Or that it reproduces on both?


Might it be Something specific to the CNI I'm using?

I'm using the `ptp` cni plugin with the `host-local` ipam with static routes between nodes. So no overlay networking  , NATing or encapsulation.

**Comment by howardjohn** [comment]
On both versions the exit code is 0 and the commands it runs are exactly identical. I was worrying we somehow misran some ipv4 vs ipv6 commands between the versions, but at least in my env that was not the case.

It might be helpful to just run a pod without Istio injected and do some ad hoc iptables commands to try to diagnose further. it looks like its entirely failing on `iptables -t nat -N ISTIO_INBOUND` (if I am reading the obscure errors from the kernel correctly), which seems odd. I am not sure why that could change between versions

**Comment by arianvp** [comment]
Running on `1.11` there _are_ some errors but they're not leading to the command failing. Wondering if that points to a cause

```
kubectl logs -lapp=nginx -c istio-init --tail 1000000
2022-02-18T16:25:26.662321Z     info    Istio iptables environment:
ENVOY_PORT=
INBOUND_CAPTURE_PORT=
ISTIO_INBOUND_INTERCEPTION_MODE=
ISTIO_INBOUND_TPROXY_ROUTE_TABLE=
ISTIO_INBOUND_PORTS=
ISTIO_OUTBOUND_PORTS=
ISTIO_LOCAL_EXCLUDE_PORTS=
ISTIO_EXCLUDE_INTERFACES=
ISTIO_SERVICE_CIDR=
ISTIO_SERVICE_EXCLUDE_CIDR=
ISTIO_META_DNS_CAPTURE=
2022-02-18T16:25:26.662366Z     info    Istio iptables variables:
PROXY_PORT=15001
PROXY_INBOUND_CAPTURE_PORT=15006
PROXY_TUNNEL_PORT=15008
PROXY_UID=1337
PROXY_GID=1337
INBOUND_INTERCEPTION_MODE=REDIRECT
INBOUND_TPROXY_MARK=1337
INBOUND_TPROXY_ROUTE_TABLE=133
INBOUND_PORTS_INCLUDE=*
INBOUND_PORTS_EXCLUDE=15090,15021,15020
OUTBOUND_IP_RANGES_INCLUDE=*
OUTBOUND_IP_RANGES_EXCLUDE=
OUTBOUND_PORTS_INCLUDE=
OUTBOUND_PORTS_EXCLUDE=
KUBEVIRT_INTERFACES=
ENABLE_INBOUND_IPV6=true
DNS_CAPTURE=false
CAPTURE_ALL_DNS=false
DNS_SERVERS=[],[]
OUTPUT_PATH=
NETWORK_NAMESPACE=
CNI_MODE=false
EXCLUDE_INTERFACES=

2022-02-18T16:25:26.662379Z     info    Running command: ip -6 addr add ::6/128 dev lo
2022-02-18T16:25:26.664707Z     info    Writing following contents to rules file: /tmp/iptables-rules-1645201526664508404.txt789499006
* nat
-N ISTIO_INBOUND
-N ISTIO_REDIRECT
-N ISTIO_IN_REDIRECT
-N ISTIO_OUTPUT
-A ISTIO_INBOUND -p tcp --dport 15008 -j RETURN
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
-A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-ports 15006
-A PREROUTING -p tcp -j ISTIO_INBOUND
-A ISTIO_INBOUND -p tcp --dport 22 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15090 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15021 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15020 -j RETURN
-A ISTIO_INBOUND -p tcp -j ISTIO_IN_REDIRECT
-A OUTPUT -p tcp -j ISTIO_OUTPUT
-A ISTIO_OUTPUT -o lo -s 127.0.0.6/32 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --uid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
COMMIT
2022-02-18T16:25:26.664754Z     info    Running command: iptables-restore --noflush /tmp/iptables-rules-1645201526664508404.txt789499006
2022-02-18T16:25:26.678354Z     info    Writing following contents to rules file: /tmp/ip6tables-rules-1645201526678234467.txt113374661
* nat
-N ISTIO_INBOUND
-N ISTIO_REDIRECT
-N ISTIO_IN_REDIRECT
-N ISTIO_OUTPUT
-A ISTIO_INBOUND -p tcp --dport 15008 -j RETURN
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
-A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-ports 15006
-A PREROUTING -p tcp -j ISTIO_INBOUND
-A ISTIO_INBOUND -p tcp --dport 22 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15090 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15021 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15020 -j RETURN
-A ISTIO_INBOUND -p tcp -j ISTIO_IN_REDIRECT
-A OUTPUT -p tcp -j ISTIO_OUTPUT
-A ISTIO_OUTPUT -o lo -s ::6/128 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d ::1/128 -m owner --uid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d ::1/128 -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d ::1/128 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
COMMIT
2022-02-18T16:25:26.678422Z     info    Running command: ip6tables-restore --noflush /tmp/ip6tables-rules-1645201526678234467.txt113374661
2022-02-18T16:25:26.679976Z     error   Command error output: ip6tables-restore v1.8.4 (legacy): ip6tables-restore: unable to initialize table 'nat'

Error occurred at line: 1
Try `ip6tables-restore -h' or 'ip6tables-restore --help' for more information.

2022-02-18T16:25:26.680024Z     info    Running command: iptables-save 
2022-02-18T16:25:26.681818Z     info    Command output: 
# Generated by iptables-save v1.8.4 on Fri Feb 18 16:25:26 2022
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
:ISTIO_INBOUND - [0:0]
:ISTIO_IN_REDIRECT - [0:0]
:ISTIO_OUTPUT - [0:0]
:ISTIO_REDIRECT - [0:0]
-A PREROUTING -p tcp -j ISTIO_INBOUND
-A OUTPUT -p tcp -j ISTIO_OUTPUT
-A ISTIO_INBOUND -p tcp -m tcp --dport 15008 -j RETURN
-A ISTIO_INBOUND -p tcp -m tcp --dport 22 -j RETURN
-A ISTIO_INBOUND -p tcp -m tcp --dport 15090 -j RETURN
-A ISTIO_INBOUND -p tcp -m tcp --dport 15021 -j RETURN
-A ISTIO_INBOUND -p tcp -m tcp --dport 15020 -j RETURN
-A ISTIO_INBOUND -p tcp -j ISTIO_IN_REDIRECT
-A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-ports 15006
-A ISTIO_OUTPUT -s 127.0.0.6/32 -o lo -j RETURN
-A ISTIO_OUTPUT ! -d 127.0.0.1/32 -o lo -m owner --uid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT ! -d 127.0.0.1/32 -o lo -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-A ISTIO_OUTPUT -j ISTIO_REDIRECT
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
COMMIT
# Completed on Fri Feb 18 16:25:26 2022

2022-02-18T16:25:26.681844Z     info    Running command: ip6tables-save 
```

**Comment by howardjohn** [comment]
Ah. I wonder if there is a bug in 1.11 that it doesn't fail hard. And then we only have two modes (ipv4 or ipv4+ipv6), so on ipv6 only kernels it fails hard now

**Comment by arianvp** [comment]
Just to note: 

I have Ipv4 on the host. (This is still useful for things like fetching containers from dockerhub which doesn't support ipv6 yet). The kubernetes cluster itself just doesn't have dual stack enabled and is only configured with an ipv6 cluster cidr. So pods only get assigned ipv6 addresses.

**Comment by bianpengyuan** [comment]
Probably the same issue as https://github.com/istio/istio/issues/36961 which is probably fixed by https://github.com/istio/istio/pull/37164.

**Comment by arianvp** [comment]
Is there an easy way to test out commits that aren't in a release yet? Happy to give it a try on my cluster to see if it fixes this

**Comment by howardjohn** [comment]
https://github.com/istio/istio/wiki/Dev%20Builds. So that was merged in
master so you would want `latest` or 1.14-dev

**Comment by arianvp** [comment]
Can confirm this is fixed on 1.14-dev! 🎊
